rpcserver.go

```diff
@@ -7637,6 +7637,29 @@ func (r *rpcServer) AddInvoice(ctx context.Context,
 	if len(req.AssetId) != sha256.Size {
 		return nil, fmt.Errorf("asset ID must be 32 bytes")
 	}
+
+	// Let's make sure that only one type of amount is set, in order to
+	// avoid ambiguous behavior. This field dictates the actual value of the
+	// invoice so let's be strict and only allow one possible value to be
+	// set.
+	if req.AssetAmount > 0 && (req.InvoiceRequest.ValueMsat > 0 ||
+		req.InvoiceRequest.Value > 0) {
+
+		return nil, fmt.Errorf("cannot set both asset amount and sats" +
+			"amount")
+	}
+
+	if req.InvoiceRequest.ValueMsat > 0 && req.InvoiceRequest.Value > 0 {
```

Comment:
We should use `lnrpc.UnmarshallAmt()`, since it already does all this validation and conversion. That way we also know we're using the same rules as the actual `lnd` backend the payment request will be handed to.